tools/tests/test_osqueryd.py

```diff
@@ -77,6 +77,37 @@ def test_3_daemon_with_watchdog(self):
         self.assertTrue(daemon.isDead(children[0]))
 
     @test_base.flaky
+    def test_3_daemon_lost_worker(self):
+        # Test that killed workers are respawned by the watcher
+        if os.environ.get('SANITIZE') is not None:
+            return
+        daemon = self._run_daemon({
+            "allow_unsafe": True,
+            "disable_watchdog": False,
+            "ephemeral": True,
+            "disable_database": True,
+            "disable_logging": True,
+        })
+        self.assertTrue(daemon.isAlive())
+
+        # Check that the daemon spawned a child process
+        children = daemon.getChildren()
+        self.assertTrue(len(children) > 0)
+
+        # Kill only the child worker
+        os.kill(children[0], signal.SIGINT)
+        self.assertTrue(daemon.isDead(children[0]))
+        self.assertTrue(daemon.isAlive())
+
```

Comment:
This depends on the test checking isDead() before the watcher respawns the worker right? Is there any way to 'pause' the watcher to make this less flake-prone?